python/shotgun_fields/percent_widget.py

```diff
@@ -8,27 +8,81 @@
 # agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
 # not expressly granted therein are reserved by Shotgun Software Inc.
 
-"""
-Widget that represents the value of a percent field in Shotgun
-"""
 import locale
+
+from sgtk.platform.qt import QtGui, QtCore
 from .label_base_widget import LabelBaseWidget
 from .shotgun_field_meta import ShotgunFieldMeta
 
 
 class PercentWidget(LabelBaseWidget):
     """
-    Inherited from a :class:`~LabelBaseWidget`, this class is able to
-    display a percent field value as returned by the Shotgun API.
+    Display a ``percent`` field value as returned by the Shotgun API.
     """
     __metaclass__ = ShotgunFieldMeta
-    _FIELD_TYPE = "percent"
+    _DISPLAY_TYPE = "percent"
 
     def _string_value(self, value):
         """
         Convert the Shotgun value for this field into a string
 
         :param value: The value to convert into a string
-        :type value: Float or Integer
+        :type value: :obj:`int` or :obj:`float`
         """
         return locale.format("%d", value, grouping=True) + "%"
+
+
+class PercentEditorWidget(QtGui.QSpinBox):
+    """
+    Allows editing of a ``percent`` field value as returned by the Shotgun API.
+
+    Pressing ``Enter`` or ``Return`` when the widget has focus will cause the
+    value to be applied and the ``value_changed`` signal to be emitted.
+    """
+    __metaclass__ = ShotgunFieldMeta
+    _EDITOR_TYPE = "percent"
+
+    def get_value(self):
+        """
+        :return: The internal value being displayed by the widget.
+        """
+        return self.value()
+
+    def keyPressEvent(self, event):
```

Comment:
rob asked the same...

> These particular "apply" key shortcuts show up in 6 or so of the line edit based widgets. Other, text edit based widgets, use different keys. Other widgets don't use key events to apply values. To me it feels like there's not enough similarity across all the widgets to put something in the metaclass. I'm happy to though if you have something you think is reasonable.

Thinking more about it, we could add an intermediate `QLineEdit` base class that has this logic. But then there are the date/datetime widgets that have their own internal `QLineEdit`. I'm inclined to leave it for now since the code is pretty minimal. Let me know if you think otherwise.